e2e/tests/uniswap_trade_test.rs

```diff
@@ -0,0 +1,186 @@
+use contracts::{ERC20Mintable, GPv2Settlement, UniswapV2Factory, UniswapV2Router02};
+use ethcontract::prelude::{Account, Address, Http, PrivateKey, Web3, U256};
+use model::{DomainSeparator, OrderCreationBuilder, OrderKind};
+use orderbook::orderbook::OrderBook;
+use secp256k1::SecretKey;
+use serde_json::json;
+use std::{str::FromStr, sync::Arc};
+use web3::signing::SecretKeyRef;
+
+const TRADER_A_PK: [u8; 32] = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
+];
+const TRADER_B_PK: [u8; 32] = [
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2,
+];
+
+const NODE_HOST: &str = "http://localhost:8545";
+const API_HOST: &str = "http://localhost:8080";
+const ORDER_PLACEMENT_ENDPOINT: &str = "/api/v1/orders/";
+
+#[tokio::test]
+async fn test_with_ganache() {
```

Comment:
We should set up tracing in this test to get log messages. Same way as in the binaries with `tracing_setup::initialize`.